src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,88 @@
+import { useMemo, useState } from 'react';
+import classNames from 'classnames';
+import debounce from 'lodash.debounce';
+import { Person } from '../../types/Person';
+
+interface Props {
+  people: Person[];
+  setPerson?:(person:Person) => void;
+  delay: number;
+}
+
+export const Autocomplete: React.FC<Props> = ({
+  people,
+  setPerson = () => {},
+  delay,
+}) => {
+  const [dropdownActive, setDropdownActive] = useState(false);
+  const [query, setQuery] = useState('');
+  const [choosePeople, setChoosePeople] = useState('');
+
+  const applyQuery = useMemo(
+    () => debounce(setQuery, delay), [delay],
+  );
+  const selectPerson = (
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+    person: Person,
+  ) => {
+    event.preventDefault();
+
+    setChoosePeople(person.name);
+    setPerson(person);
+    setDropdownActive(false);
+  };
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setChoosePeople(e.target.value);
+    applyQuery(e.target.value);
+  };
+
+  const peopleFilter = useMemo(() => {
+    return people
+      .filter(person => person.name.toLocaleLowerCase()
+        .includes(query.toLocaleLowerCase()));
+  }, [query, people]);
+
+  return (
+    <div className={classNames('dropdown', {
+      ' is-active': dropdownActive,
+    })}
+    >
+      <div className="dropdown-trigger ">
+        <input
+          type="text"
+          value={choosePeople}
+          aria-haspopup="true"
+          placeholder="Enter a part of the name"
+          className="input"
+          onChange={handleInputChange}
+          onFocus={() => setDropdownActive(true)}
+        />
+      </div>
+
+      <div className="dropdown-menu" role="menu">
+        <div
+          className="dropdown-content"
+          style={{ overflow: 'auto', height: '300px' }}
```

Comment:
it's a bad practice for add inline styles in a template, it's better to move this styles to css file